lib/apiservers/engine/backends/container.go

```diff
@@ -506,3 +653,235 @@ func (c *Container) imageExist(imageID string) (storeName string, err error) {
+				// current Docker behavior?
+				stdinReader.Close()
+			} else {
+				// Shutdown the client's request for stdout, stderr
+				errCancel()
+				outCancel()
+			}
+			wg.Done()
+		}()
+	}
+
+	if clStdout != nil {
+		transport := httptransport.New(PortLayerServer(), "/", []string{"http"})
+		plClient := client.New(transport, nil)
+		transport.Consumers["application/octet-stream"] = httpkit.ByteStreamConsumer()
+		transport.Producers["application/octet-stream"] = httpkit.ByteStreamProducer()
+
+		// Swagger -> pipewriter.  Synchronous, blocking call
+		go func() {
+			getStdoutParams := interaction.NewContainerGetStdoutParamsWithContext(outContext).WithID(name)
+			_, err := plClient.Interaction.ContainerGetStdout(getStdoutParams, clStdout)
+			if clStdin != nil {
+				// Close the client stdin connection (e.g. CLI)
+				clStdin.Close()
+				inCancel()
+			}
+			if err != nil {
+				if _, ok := err.(*interaction.ContainerGetStdoutNotFound); ok {
+					errors <- derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", name))
+					wg.Done()
+					return
+				}
+
+				// If we get here, most likely something went wrong with the port layer API server
+				errors <- derr.NewErrorWithStatusCode(fmt.Errorf("Unknown error from the interaction port layer: %s", err),
+					http.StatusInternalServerError)
+				wg.Done()
+				return
+			}
+
+			errors <- nil
+			wg.Done()
+		}()
+	}
+
+	if clStderr != nil {
+		transport := httptransport.New(PortLayerServer(), "/", []string{"http"})
+		plClient := client.New(transport, nil)
+		transport.Consumers["application/octet-stream"] = httpkit.ByteStreamConsumer()
+		transport.Producers["application/octet-stream"] = httpkit.ByteStreamProducer()
+
+		// Swagger -> pipewriter.  Synchronous, blocking call
+		go func() {
+			getStderrParams := interaction.NewContainerGetStderrParamsWithContext(errContext).WithID(name)
+			_, err := plClient.Interaction.ContainerGetStderr(getStderrParams, clStderr)
+			if clStdin != nil {
+				// Close the client stdin connection (e.g. CLI)
+				clStdin.Close()
+				inCancel()
+			}
+			if err != nil {
+				if _, ok := err.(*interaction.ContainerGetStderrNotFound); ok {
+					errors <- derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", name))
+					wg.Done()
+					return
+				}
+
+				// If we get here, most likely something went wrong with the port layer API server
+				errors <- derr.NewErrorWithStatusCode(fmt.Errorf("Unknown error from the interaction port layer: %s", err),
+					http.StatusInternalServerError)
+				wg.Done()
+				return
+			}
+
+			errors <- nil
+			wg.Done()
+		}()
+	}
+
+	// Wait for all stream copy to exit
+	wg.Wait()
+	log.Printf("Attach stream closed")
+	defer close(errors)
+	for err := range errors {
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Code c/c from io.Copy() modified by Docker to handle escape sequence
+func copyEscapable(dst io.Writer, src io.ReadCloser, keys []byte) (written int64, err error) {
```

Comment:
Can we do this at a later date?  Our copyright script doesn't have an ignore function.  I separated the file, but the build fails on copyright check.  I don't want to add vmware's copyright to this file, and I don't want to make the changes to the copyright script as part of this PR.